scopus-import/src/main/java/no/sikt/nva/scopus/conversion/NvaCustomerConnection.java

```diff
@@ -30,8 +31,23 @@ public NvaCustomerConnection(AuthorizedBackendUriRetriever uriRetriever) {
         this.uriRetriever = uriRetriever;
     }
 
-    public boolean isNvaCustomer(CristinOrganization cristinOrganization) {
-        return isPresentAndHasTopLevelOrg(cristinOrganization) && isCustomer(cristinOrganization);
+    public NvaCustomer fetchCustomer(CristinOrganization cristinOrganization) {
+        return Optional.ofNullable(cristinOrganization)
+                   .filter(NvaCustomerConnection::hasTopLevelOrg)
+                   .filter(this::isCustomer)
+                   .map(org -> new NvaCustomer(true, getCristinTopLevelOrg(cristinOrganization)))
+                   .orElse(defaultNvaCustomer());
+    }
```

Comment:
We can talk about caching something here…I don't think it's necessary to repeat the call…